### PR TITLE
Add support for more environment variables

### PR DESCRIPTION
**Pull request by delner**
This pull request adds global tags for:

- `DD_TAGS` (acts as base set of global tags, overridden by other tags)
- `DD_ENV` --> `env:<env name>`
- `DD_SERVICE` --> `service:<service name>`
- `DD_VERSION` --> `version:<version string>`

Precedence in order from most to least preferred is manually defined global tags (via `#initialize`), `DD_ENV`/`DD_SERVICE`/`DD_VERSION`, then finally `DD_TAGS`.

### CODE REVIEW

**Comment by delner** [comment]
@remeh Okay, there's one more question I had... if we set these global tags (say `env:staging`), and the user passes `statsd.count('hits', 1, tags: ['env:staging'])`, will this produce duplicate `env:staging` tags? If so would we want to remove this kind of duplication?

Just thinking of the trade-off between saving on packet size vs saving on memory/CPU it would take to de-dup this.

**Comment by kbogtob** [comment]
Thanks @delner for your contribution! 
I've just fixed some nits to homogenise the code with the existing one and added some tests.

It's true that we don't deduplicate the tags and I think we should be able to override specific tags in this order: ENV < Global tags < Call tags. I will discuss it internally and create an issue.

**Comment by delner** [comment]
@kbogtob Changes seem fine to me. Also just added a minor fix for a warning that was blowing up our test suite (when we'd close Statsd without using it.)

File: lib/datadog/statsd/serialization/tag_serializer.rb [hunk 1]
```diff
@@ -28,11 +30,33 @@ def format(message_tags)
...
         private
+
+        def to_tags_hash(tags)
+          case tags
+          when Hash
+            tags.dup
+          when Array
+            Hash[
+              tags.collect do |string|
+                string.split(':').tap do |tokens|
```
Comment (delner) [inline lib/datadog/statsd/serialization/tag_serializer.rb:41]: There's a bug here... doesn't handle `':'` well as input; looking into it.

Comment (delner) [reply lib/datadog/statsd/serialization/tag_serializer.rb]: Okay this should be fixed now.

File: lib/datadog/statsd/serialization/tag_serializer.rb [hunk 2]
```diff
@@ -28,11 +30,33 @@ def format(message_tags)
...
 
         private
+
+        def to_tags_hash(tags)
+          case tags
+          when Hash
+            tags.dup
+          when Array
+            Hash[
+              tags.collect do |string|
```
Comment (kbogtob) [inline lib/datadog/statsd/serialization/tag_serializer.rb:40]: Could you please use `map`? the project uses `map` over `collect` :)

**Pull request merged**